atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingIteratorTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import static com.palantir.atlasdb.transaction.impl.expectations.TrackingIteratorTestUtils.STRING_MEASURER;
+import static com.palantir.atlasdb.transaction.impl.expectations.TrackingIteratorTestUtils.consumeIteratorIntoList;
+import static com.palantir.atlasdb.transaction.impl.expectations.TrackingIteratorTestUtils.createStringIterator;
+import static com.palantir.atlasdb.transaction.impl.expectations.TrackingIteratorTestUtils.noOp;
+import static org.assertj.core.api.Assertions.assertThatIterable;
+import static org.assertj.core.api.Assertions.assertThatIterator;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public final class TrackingIteratorTest {
+    @Test
+    public void trackingIteratorForwardsData() {
+        TrackingIterator<String, Iterator<String>> trackingIterator =
+                new TrackingIterator<>(createStringIterator(), noOp(), STRING_MEASURER);
+
+        assertThatIterator(trackingIterator)
+                .toIterable()
+                .containsExactlyElementsOf(consumeIteratorIntoList(createStringIterator()));
```

Comment:
No need for `consumeIteratorIntoList`, you can just use `ImmutableList.copyOf`!

Reply:
You can also just reference the underlying list the iterator is backed by -- imo both are acceptable.

Reply:
Also we can just use `assertThat` which will delegate to `assertThatIterator` under the hood